### ISSUE TITLE
Gentzen sequent calculus?

### ISSUE THREAD

**Issue by vcvpaiva** [Carnap/Carnap#3]
sorry this is a question, do we have it? where? thanks!

**Comment by gleachkr** [comment]
We don't have a way of constructing sequent calculus deductions at the moment. 

We do have sequents, [here](https://github.com/gleachkr/Carnap/tree/master/Carnap/src/Carnap/Languages/ClassicalSequent), which are used in proof-checking for natural deduction and the representation of inference-schemes. So, e.g. we give the scheme for modus ponens as:

```haskell
modusPonens :: BooleanRule lex b
modusPonens = [ GammaV 1 :|-: SS (phin 1 .→. phin 2)
              , GammaV 2 :|-: SS (phin 1)
              ] ∴ GammaV 1 :+: GammaV 2 :|-: SS (phin 2)
```

The place where we'd want to add something handling deductions in the sequent calculus would be under https://github.com/gleachkr/Carnap/tree/master/Carnap/src/Carnap/Calculi. Some work would also need to be done on the front end for manipulating them in a natural way.

I'll be teaching a second-semester major-level logic class next year, though, and I'd really like to include Gentzen's LK and LJ. So I'm hoping I'll be able to implement support for this over the summer.

**Comment by gleachkr** [comment]
Very pleased to report that this is finally implemented, as of bdd8f0f081e3d432e78010c4c87219767af2c5f0. You can see it in action in example 14 of the about page at carnap.io/about. 

Documentation is underway. Although functional, it's probably still a little rough around the edges. I'll also continue to improve the interface, which is mostly implemented in pure JavaScript, at https://github.com/gleachkr/ProofJS.
